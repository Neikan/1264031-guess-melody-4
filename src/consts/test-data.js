const AVATAR_URL = `https://api.adorable.io/avatars/128`;
const SRC_URL = `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`;


export const GameType = {
  WELCOME: `welcome`,
  ARTIST: `artist`,
  GENRE: `genre`,
};


export const questionGenre = {
  aspect: GameType.GENRE,
  genre: `rock`,
  answers: [
    {
      id: `song-1`,
      src: SRC_URL,
      genre: `rock`,
    }, {
      id: `song-2`,
      src: SRC_URL,
      genre: `blues`,
    }, {
      id: `song-3`,
      src: SRC_URL,
      genre: `jazz`,
    }, {
      id: `song-4`,
      src: SRC_URL,
      genre: `rock`,
    }
  ]
};


export const questionArtist = {
  aspect: GameType.ARTIST,
  song: {
    artist: `Jim Beam`,
    src: SRC_URL,
  },
  answers: [
    {
      id: `artist-1`,
      picture: `${AVATAR_URL}/0`,
      artist: `John Snow`,
    }, {
      id: `artist-2`,
      picture: `${AVATAR_URL}/1`,
      artist: `Jack Daniels`,
    }, {
      id: `artist-3`,
      picture: `${AVATAR_URL}/2`,
      artist: `Jim Beam`,
    }
  ]
};
