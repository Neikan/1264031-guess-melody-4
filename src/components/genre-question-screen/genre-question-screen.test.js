import React from 'react';
import renderer from 'react-test-renderer';
import GenreQuestionScreen from './genre-question-screen.jsx';
import {questionGenreScreen} from '../../consts/test-data.js';


describe(`Test ArtistQuestionScreen component`, () => {
  test(`GenreQuestionScreen is rendered correctly`, () => {
    const tree = renderer.create((
      <GenreQuestionScreen
        question = {questionGenreScreen}
        onAnswerChange = {() => {}}
      />
    ), {
      createNodeMock: () => {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
