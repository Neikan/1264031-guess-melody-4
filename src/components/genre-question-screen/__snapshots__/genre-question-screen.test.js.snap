// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ArtistQuestionScreen component GenreQuestionScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    rock
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="song-1"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="song-1"
        />
        <label
          className="game__check"
          htmlFor="song-1"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="song-2"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="song-2"
        />
        <label
          className="game__check"
          htmlFor="song-2"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="song-3"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="song-3"
        />
        <label
          className="game__check"
          htmlFor="song-3"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="song-4"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="song-4"
        />
        <label
          className="game__check"
          htmlFor="song-4"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
